new-packages/ts-project/src/state.ts

```diff
@@ -157,6 +182,65 @@ export function extractState(
           });
           break;
         }
+        case 'entry':
+        case 'exit': {
+          const blocks = mapMaybeArrayElements(
+            ts,
+            prop.initializer,
+            (element): ActionBlock | undefined => {
+              if (isUndefined(ts, element)) {
+                return;
+              }
+              if (ts.isStringLiteralLike(element)) {
+                return createActionBlock({
+                  sourceId: element.text,
+                  parentId: node.uniqueId,
+                });
+              }
+              if (ts.isObjectLiteralExpression(element)) {
+                const typeProperty = element.properties.find(
+                  (prop): prop is PropertyAssignment =>
+                    ts.isPropertyAssignment(prop) &&
+                    getPropertyKey(ctx, ts, prop) === 'type',
+                );
+
+                if (!typeProperty) {
+                  return;
+                }
+
+                if (ts.isStringLiteralLike(typeProperty.initializer)) {
+                  return createActionBlock({
+                    sourceId: typeProperty.initializer.text,
+                    parentId: node.uniqueId,
+                  });
+                }
+              }
+
+              return createActionBlock({
+                sourceId: `inline:${uniqueId()}`,
+                parentId: node.uniqueId,
+              });
+            },
+          );
+
+          if (!everyDefined(blocks)) {
+            ctx.errors.push({
+              type: 'state_property_unhandled',
+            });
+            continue;
+          }
```

Comment:
So for now, we'd lose all the actions in entry for example if one of them can't be extracted?

Reply:
Yes, that was what we agreed on yesterday in the evening. While it's not ideal, it won't require any immediate changes in the Studio.